cdap-api/src/main/java/co/cask/cdap/api/TaskLocalizationContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Localization context for Program Runtimes (MapReduce Tasks/Spark Executors).
+ */
+@Beta
+public interface TaskLocalizationContext {
+
+  /**
+   * @param name the local file's name
+   * @return a {@link File} for the localized file
+   */
+  File getLocalFile(String name) throws IOException;
+
+  /**
+   *
+   * @param name the local file's name
+   * @param archive indicates if the requested file is an archive
+   * @return a {@link File} for the localized file
+   */
+  File getLocalFile(String name, boolean archive) throws IOException;
```

Comment:
But for example, we need to decide whether to call ``Job.getLocalCacheFiles()`` or ``Job.getLocalCacheArchives`` in the case of map-reduce, right? Or are you saying always call ``Job.getLocalCacheFiles()``?